Yoast/wordpress-seo
# Remove Facebook App ID support

## Context


* Removes support for the Facebook App ID.

## Summary


This PR can be summarized in the following changelog entry:

* Removes support for the Facebook App ID.

## Relevant technical choices:

* Removes the Facebook App ID field from the SEO -> Social -> Facebook page in the admin.
* Deprecates the `validate_facebook_app_id_api_response_code` and `validate_facebook_app_id_api_response_body` filters.
* Deprecates the `validate_facebook_app_id` and `generate_open_graph_fb_app_id` methods. Both methods are also no longer being called anywhere in the code base.
* Deprecates the `FB_App_ID_Presenter` class and no longer automatically loads it.
* Removes the `fbadminapp` value from the `wpseo_social` option.

## Test instructions

### Test instructions for the acceptance test before the PR gets merged
This PR can be acceptance tested by following these steps:

* Ensure you have the Yoast Test Helper installed and activated.
* Ensure you have a Facebook App ID set up in SEO -> Social -> Facebook. One way to do this without having to register an app with Facebook, is to navigate to the `wp_options` database table, search for the `wpseo_social` option and ensure that `fbadminapp` exists. If it does, ensure you set a value (i.e. `s:10:"fbadminapp";s:1:"1";`) or use the pasted value below as a replacement for your own:

`a:19:{s:13:"facebook_site";s:0:"";s:13:"instagram_url";s:0:"";s:12:"linkedin_url";s:0:"";s:11:"myspace_url";s:0:"";s:16:"og_default_image";s:0:"";s:19:"og_default_image_id";s:0:"";s:18:"og_frontpage_title";s:0:"";s:17:"og_frontpage_desc";s:0:"";s:18:"og_frontpage_image";s:0:"";s:21:"og_frontpage_image_id";s:0:"";s:9:"opengraph";b:1;s:13:"pinterest_url";s:0:"";s:15:"pinterestverify";s:0:"";s:7:"twitter";b:1;s:12:"twitter_site";s:0:"";s:17:"twitter_card_type";s:19:"summary_large_image";s:11:"youtube_url";s:0:"";s:13:"wikipedia_url";s:0:"";s:10:"fbadminapp";s:3:"123";}`

* Check the source of a post or a page and ensure the `<meta property="fb:app_id" content="123">` tag is present.
* Checkout this branch, build it and use the test helper to set the version to `15.3-RC0`.
* Refresh the post/page you were looking at in the previous step. Ensure that the meta property no longer exists.
* Refresh the database table and ensure the `fbadminapp` value no longer exists.

#### Testing filter deprecations
* Ensure that `define( 'WP_DEBUG', true );` is present in your WordPress' configuration file (i.e. `config/basic-wordpress-config.php` if you use the [plugin-development-docker](https://github.com/Yoast/plugin-development-docker) environment ).
* Edit the `functions.php` file of your active theme and add the following code:
```
add_filter( 'validate_facebook_app_id_api_response_code', function() {} );
add_filter( 'validate_facebook_app_id_api_response_body', function() {} );
```
* Navigate to the backend of your testing environment.
  * If you **do not** have the Query Monitor plugin active, you should see the deprecation warnings at the top of your screen.
  * If you do have the Query Monitor plugin active, you should be able to find the deprecation warnings under the PHP Errors section within Query Monitor.

**Please note**: Setting `WP_DEBUG` to `false` will hide said deprecation warnings.

### Test instructions for QA when the code is in the RC




QA can test this PR by following these steps:

*

## Impact check

This PR affects the following parts of the plugin, which may require extra testing:

*

## UI changes


## Documentation

* [x] I have written documentation for this change. See https://github.com/Yoast/developer-docs/pull/108

## Quality assurance


Fixes [P3-265]
